edu.cuny.hunter.log.core/src/edu/cuny/hunter/log/core/refactorings/LogRejuvenatingProcessor.java

```diff
@@ -314,6 +315,20 @@ public RefactoringStatus checkFinalConditions(final IProgressMonitor monitor, fi
 		return status;
 	}
 
+	/**
+	 * Given a set of log invocations, return a set of enclosing methods.
+	 * 
+	 * @return a set of enclosing methods
+	 */
+	private HashSet<IMethod> getEnclosingMethods() {
+		HashSet<IMethod> methods = new HashSet<IMethod>();
+		this.logInvocationSet.parallelStream().forEach(inv -> {
```

Comment:
Use a filter and collector here. See https://docs.oracle.com/javase/10/docs/api/java/util/stream/package-summary.html.